TACC/tup-ui
# feat(tup-ui): dashboard layout

## Overview

Lay out the dashboard.

## Related

- [TUP-316](https://jira.tacc.utexas.edu/browse/TUP-316)

## Changes

Difficult to review until #92 is merged.

- added global section space variables (like CEPv2 does)
- added Dashboard layout styles
- changed Dashboard to use a `<SectionHeader>` (but **not** `<Section>`)
- removed `<Section>` CSS flex code
- changed default `<SectionContent>` layout to `one-column`
- changed `<PageLayout>` to allow scrolling of panels

## Testing

[Deployed to dev.tup on 2022-12-20.](https://tacc-team.slack.com/archives/C02GF2X2AS3/p1671571908970539)

1. View the Dashboard on a screen ≥1200px wide.
1. View the Dashboard on a screen <992px wide.
1. View the Dashboard on a screen ≥900px tall.
1. View the Dashboard on a screen <900px tall.
2. Verify dashboard content is navigable.*
    <sup>* Except for System Monitor, which should abandon `<InfiniteScrollTable>`.</sup>
3. Verify layout solutions resemble CEPv2's.


## UI

### 1200px Wide

This solution (mimicked from CEPv2) avoids styling short viewport by forcing scrollbar when content gets crowded.

| >900px | <900px (scroll up) | <900px (scroll down) |
| - | - | - |
| ![TUP-316 1200 x  900](https://user-images.githubusercontent.com/62723358/208547251-cb3c6fb5-9fb6-46ed-9ead-d720269e60bb.png) | ![TUP-316 1200 x  900 - Up](https://user-images.githubusercontent.com/62723358/208547250-f7122b93-52f2-4fc0-b10a-d343bbd94c2c.png) | ![TUP-316 1200 x  900 - Down](https://user-images.githubusercontent.com/62723358/208547247-347b06ed-2e54-456f-9360-ab7613e46d2a.png) |

### 992px Wide

This solution (from CEPv2 but via CSS grid) stacks panels atop one another on narrow screens.

| >900px | <900px (scroll up) | <900px (scroll down) |
| - | - | - |
![TUP-316 992 x  900](https://user-images.githubusercontent.com/62723358/208547245-322ba096-e068-4a30-bce4-efe83dd91958.png) | ![TUP-316 992 x  900 - Up](https://user-images.githubusercontent.com/62723358/208547242-b7ff2b56-42dd-4502-a73d-59bd54f42368.png) | ![TUP-316 992 x  900 - Down](https://user-images.githubusercontent.com/62723358/208547241-cd41cc81-0317-4cf4-86db-048622419591.png) |

## Notes

### Tasks

- [x] ⚠️ Merge #92.
- [x] 💡 Replace Sections (they seem obtuse and overkill for tup-ui).
